### ISSUE TITLE
Предложение. Добавить возможность обработки текста до записи в базу и хранить оригинальный текст в отдельном поле.

### ISSUE THREAD

**Issue by ivan-cc** [tizis/laravel-comments#4]
Сейчас текст комментария жёстко правится методом `\tizis\laraComments\UseCases\CommentService::htmlFilter` и в таком виде сохраняется.

Например, я хочу использовать смайлики или markdown-разметку, сейчас мне для этого можно использовать только [постпроцессинг](https://github.com/tizis/laravel-comments/blob/master/readme-ru.md#api-%D0%BF%D1%80%D0%B5%D0%BF%D1%80%D0%BE%D1%86%D0%B5%D1%81%D1%81%D0%B8%D0%BD%D0%B3) при каждом отображении комментария, но с т.ч. производительности это плохое решение.

Есть пару предложений:
1. Дать возможность использования своего аналога `htmlFilter` (например, тоже указывать его в настройках), чтобы можно было обработать текст комментария своей логикой.
2. Сделать дополнительное поле в БД (`original_comment`) и хранить оригинальный текст комментария в этом поле, чтобы показывать его содержимое во время редактирования комментария.

**Comment by iooe** [comment]
Да, мне это тоже не очень нравится.

Я уже разбил функциональность пакета на контроллер/сервис/апи для предоставления людям больших возможностей (в крайнем случае можно отключить роуты по умолчанию и переопределять нужные методы контроллере) 
В общем то и препроцессинг был введен еще до того, как сервис стал полностью публично доступен. 

Просто проблема в том, что этот пакет я создал для себя из-за того, что не было лучших аналогов (либо ничего нет из коробки, либо жуткие проблемы с производительностью из-за игнорирования проблемы  select+1 (но это уже не так актуально)).

Была идея разбить всю библиотеку на компоненты ядро(сервис, модели и тд) и интерфейс (роуты и контроллер), но интерес постепенно угас.

**Comment by ivan-cc** [comment]
А есть ли смысл делать очередной комбайн из системы-комментариев?

Я когда выбирал систему комментирования, остановился на этой, потому что всё просто ставилось и сразу работало.

Вот только не хватает мелкой гибкости. Может еще количество событий увеличить, чтобы можно было свои извращения прикрутить :)

**Comment by iooe** [comment]
> А есть ли смысл делать очередной комбайн из системы-комментариев?
Да не было бы комбайна, просто было все в разы удобнее) 

Текущей репозиторий можно было бы сделать в виде готовой сборки с указанием зависимостей и минимальной настройкой. По аналогии с ларавел. 

Та же система рейтинга сейчас прибита намертво, было бы неплохо и ее выделить, добавив возможность своей реализации.

**Comment by iooe** [comment]
Но библиотека не так популярно, чтобы тратить кучу времени на то, что нужно трем (в лучшем случае) людям (которые в крайнем случае могут сделать свою реализацию на основе готового ядра-сервиса).

**Comment by ivan-cc** [comment]
Ну, как понимаю, прообразом были комменты от laravelista, а у него 183 звезды, всего в три раза больше звёзд (и то это он за два года набрал).

Мне кажется, большинство людей текущая система устраивает. Наверно можно было бы еще пару распространенных фич приделать, типа анонимных комментов. Ведь даже рейтинг комментариев совсем не у всех библиотек есть, а как в 2019/20 году жить без рейтинга комментариев? ;)

Предлагаю напирать на маленькие, но востребованные правки, на мои, например :+1:

**Comment by ivan-cc** [comment]
PS. А если хочется красивого решения, да еще и популярного, то лучше написать хорошую библиотеку для работы с пользователями на сайте (авторизацию, соцсети, профиль и остальные юзер-фичи). А то у laravel с этим какая-то напряженка. Например у Yii2 есть супер-пупер компонент [yii2-usuario](https://github.com/2amigos/yii2-usuario), который ставишь и у тебя сразу и формочки все и контроллеры и вход через соцсети и "Личный кабинет пользователя" и всё это довольно гибкое, можно кастомизировать очень широко. А у Laravel, самое похожее нашёл только в виде [готового шаблона](https://github.com/jeremykenedy/laravel-auth), что совсем не красиво. А другие варианты связаны с кучей рутины по кодингу всех этих вьюшек с контроллерами.

**Comment by iooe** [comment]
> (авторизацию, соцсети, профиль и остальные юзер-фичи)

Да, как минимум год назад ничего хорошего не было. В случае социалок  так вообще ужас —  socialite с куцей документацией и без официальных примеров.

**Comment by ivan-cc** [comment]
Это было открытием для меня. Я думал, что если уж для Yii люди сделали такую красоту, то уж для laravel-то этих красот будет на каждом шагу, но нет.

Человек, который сделает хороший юзер-пакет для лары, поработит мир!
